lib/CGameState.cpp

```diff
@@ -1957,7 +1956,16 @@ BattleField CGameState::battleGetBattlefieldType(int3 tile, CRandomGenerator & r
 		*RandomGeneratorUtil::nextItem(t.terType->battleFields, rand));
 }
 
-UpgradeInfo CGameState::getUpgradeInfo(const CStackInstance &stack)
+
+void CGameState::fillUpgradeInfo(const CArmedInstance *obj, SlotID stackPos, UpgradeInfo &out) const
```

Comment:
Overload below receives reference. Here you also asserting non-zero obj, consider to use reference

Reply:
Pointless. Callers always operate with pointer here.
Sure, I can replace all calls with conversion of pointer to reference but it won't protect against caller converting nullptr into null reference.